plan/planbuilder.go

```diff
@@ -783,8 +783,21 @@ func (b *planBuilder) buildSimple(node ast.StmtNode) Plan {
 		b.visitInfo = appendVisitInfo(b.visitInfo, mysql.CreateUserPriv, "", "", "")
 	case *ast.GrantStmt:
 		b.visitInfo = collectVisitInfoFromGrantStmt(b.visitInfo, raw)
-	case *ast.SetPwdStmt, *ast.RevokeStmt, *ast.KillStmt:
+	case *ast.SetPwdStmt, *ast.RevokeStmt:
 		b.visitInfo = appendVisitInfo(b.visitInfo, mysql.SuperPriv, "", "", "")
+	case *ast.KillStmt:
+		// If you have the SUPER privilege, you can kill all threads and statements.
+		// Otherwise, you can kill only your own threads and statements.
+		sm := b.ctx.GetSessionManager()
+		if sm != nil {
+			processList := sm.ShowProcessList()
```

Comment:
Could you add an interface for SessionManger like GetProcessByConnID(connID) ?

Reply:
`GetProcessByConnID` make `SessionManager` more complex.
`GetProcessByConnID` can be implemented using `ShowProcessList` which is already in `SessionManager`
I assume `Kill` will not be too frequency so this choice would not hurt the performance.
@shenli

Reply:
So the interface `ShowProcessList` return a map instead of a list.

Reply:
Yes.